File: LeetCode/Easy/1822-sign-of-the-product-of-an-array/1822-sign-of-the-product-of-an-array.java
class Solution {
    public int arraySign(int[] nums) {

        if(nums.length == 0) return 0;

        int minus = 0;

        for (int n : nums) {
            if(n < 0) {
                minus++;
            } else if (n == 0) {
                return 0;
            }
        }

        return minus % 2 == 0 ? 1 : -1;
    }
}

File: LeetCode/Easy/1822-sign-of-the-product-of-an-array/1822-sign-of-the-product-of-an-array.kt
class Solution {
    
    fun arraySign(nums: IntArray): Int {

        if(nums.size == 0) return 0

        var minus: Int = 0

        for(n in nums) {
            if(n < 0) {
                minus++
            } else if (n == 0) {
                return 0
            }
        }

        return if (minus % 2 == 0) 1 else -1
        
    }

}